src/Microsoft.AspNet.Authentication/ClaimsTransformer.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.Authentication
+{
+    public class ClaimsTransformer : IClaimsTransformer
+    {
+        public Func<ClaimsPrincipal, Task<ClaimsPrincipal>> TransformDelegate { get; set; }
```

Comment:
Glad to see you're merging the two delegates! :+1: 

That said, this name is a bit weird. Why not `OnTransform` to match the pattern used for the events?